### PR TITLE
Make the rotation period and sizes customizable

### PR DESCRIPTION
**Pull request by mpfz0r**
While the default should be good enough,
add a new config option:

`time_size_optimizing_rotation_period = 1d`
`time_size_optimizing_rotation_min_size = 20gb`
`time_size_optimizing_rotation_max_size = 50gb`

We also convert `java.time.Period` back to `org.joda.time.Period` since it allows smaller durations
which is helpful for testing and needs less conversions

/nocl
/jenkins-pr-deps https://github.com/Graylog2/graylog-plugin-enterprise/pull/4572

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog2/configuration/ElasticsearchConfiguration.java [hunk 1]
```diff
@@ -96,6 +99,16 @@ public class ElasticsearchConfiguration {
     @Parameter(value = "elasticsearch_max_number_of_indices", required = true, validator = PositiveIntegerValidator.class)
     private int maxNumberOfIndices = 20;
 
+    // TimeBasedSizeOptimizingStrategy Rotation
+    @Parameter(value = "time_size_optimizing_rotation_period")
+    private Period timeSizeOptimizingRotationPeriod = Period.days(1);
```
Comment (patrickmann) [inline graylog2-server/src/main/java/org/graylog2/configuration/ElasticsearchConfiguration.java:104]: Do we need to make sure rotation period is a multiple of full days, or an integer division of a day? Can anything bad happen if this is changed to e.g. 7 hours?

File: graylog2-server/src/main/java/org/graylog2/indexer/IndexSetValidator.java [hunk 2]
```diff
@@ -104,31 +105,23 @@ private Violation validatePrefix(IndexSetConfig newConfig) {
     @Nullable
     public Violation validateRotation(RotationStrategyConfig rotationStrategyConfig) {
         if ((rotationStrategyConfig instanceof TimeBasedSizeOptimizingStrategyConfig smartConfig)) {
```
Comment (patrickmann) [inline graylog2-server/src/main/java/org/graylog2/indexer/IndexSetValidator.java:107]: "smartConfig" no longer matches the naming scheme

File: graylog2-server/src/main/java/org/graylog2/configuration/ElasticsearchConfiguration.java (hunk 1 above)
Comment (mpfz0r) [reply graylog2-server/src/main/java/org/graylog2/configuration/ElasticsearchConfiguration.java:104]: No, I'm currently testing it with minutes `PT1M`

**Review by patrickmann — APPROVED** [review]
LGTM - I didn't do in-depth testing, since the changes are functionally equivalent.

**Pull request merged**